x/qgb/README.md

```diff
@@ -1,50 +1,320 @@
+
+If the all the attestations in store are expired, which is an edge case that should never occur, the QGB state machine [doesn't prune](https://github.com/celestiaorg/celestia-app/blob/0629c757ef35a24187a8d7a4c706c7cdc894c8b6/x/qgb/abci.go#L161) the latest attestation.
+
+### Hooks
+
+#### Validator unbonding hook
+
+When a validator starts unbonding, a [hook](https://github.com/celestiaorg/celestia-app/blob/0629c757ef35a24187a8d7a4c706c7cdc894c8b6/x/qgb/keeper/hooks.go#L23-L34) is executed that [sets](https://github.com/celestiaorg/celestia-app/blob/0629c757ef35a24187a8d7a4c706c7cdc894c8b6/x/qgb/keeper/hooks.go#LL33C2-L33C2) the `LatestUnBondingBlockHeight` to the current block height. This allows creating a new valset that removes that validator from the valset members so that he doesn't need to sign attestations afterwards.
+
+## Events
+
+### New attestation event
+
+After creating a new attestation, and adding it to the QGB store, an event is [emitted](https://github.com/celestiaorg/celestia-app/blob/0629c757ef35a24187a8d7a4c706c7cdc894c8b6/x/qgb/keeper/keeper_attestation.go#L16-L22) containing its nonce.
+
+## Client
+
+### Query attestation command
+
+The QGB query attestation command is part of the `celestia-appd` binary. It allows the user to query specific attestations by their corresponding nonce.
+
+```shell
+$ celestia-appd query qgb attestation --help                                                     INT ✘ 
+query an attestation by nonce
+
+Usage:
+  celestia-appd query qgb attestation <nonce> [flags]
+
+Aliases:
+  attestation, att
+
+Flags:
+      --grpc-addr string   the gRPC endpoint to use for this chain
+      --grpc-insecure      allow gRPC over insecure channels, if not TLS the server must use TLS
+      --height int         Use a specific height to query state at (this can error if the node is pruning state)
+  -h, --help               help for attestation
+      --node string        <host>:<port> to Tendermint RPC interface for this chain (default "tcp://localhost:26657")
+  -o, --output string      Output format (text|json) (default "text")
+
+Global Flags:
+      --chain-id string      The network chain ID
+      --home string          directory for config and data (default "/home/midnight/.celestia-app")
+      --log-to-file string   Write logs directly to a file. If empty, logs are written to stderr
+      --log_format string    The logging format (json|plain) (default "plain")
+      --log_level string     The logging level (trace|debug|info|warn|error|fatal|panic) (default "info")
+      --trace                print out full stack trace on errors
```

Comment:
[not blocking suggestion]

imo we should be diligent in removing things that do not directly contribute to this module, we might even want to remove the other flags/aliases section.
```suggestion
```